benchexec/tools/esbmc.py

```diff
@@ -31,13 +31,20 @@ def version(self, executable):
     def name(self):
         return "ESBMC"
 
-    def cmdline(self, executable, options, tasks, propertyfile, rlimits):
+    def cmdline(self, executable, options, task, rlimits):
+        tasks = list(task.input_files_or_identifier)
         assert len(tasks) == 1, "only one inputfile supported"
         inputfile = tasks[0]
-        return [executable] + ["-p", propertyfile] + options + [inputfile]
 
-    def determine_result(self, returncode, returnsignal, output, isTimeout):
-        output = "\n".join(output)
+        data_model_param = util.get_data_model_from_task(
+            task, {"ILP32": "32", "LP64": "64"}
+        )
+        if data_model_param and data_model_param not in options:
```

Comment:
I think this is errorprone, because the `--arch` parameter is added only if there is no item with the value `32` (or `64` respectively) in the command line. Such a value could appear as content of a different parameter, though. Furthermore, if a user adds `--arch` in their benchmark definition with a different value than what this code would produce, the parameter gets added twice.

Reply:
fixed in fc874d47b6d1a031f0cc7403a0ebf0706670ceea

Now using the param name for searching in optipons